File: 61-rotate-list/61-rotate-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if(head == null) 
            return null;
        ListNode tail = head;
        int n = 1;
        while(tail.next != null) {
            n++;
            tail = tail.next;
        }
        int breakpt = n - (k%n);
        int i = 0;
        if(breakpt == n)
            return head;
        
        ListNode ptr = head, prev = head;
        while(i < breakpt) {
            i++;
            prev = ptr;
            ptr = ptr.next;
        }
        tail.next = head;
        head = ptr;
        prev.next = null;
        
        return head;
    }
}

File: 61-rotate-list/61-rotate-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def rotateRight(self, head, k):
        if head == None:
            return head
        tail = head
        n = 1
        
        while tail.next != None:
            n += 1
            tail = tail.next
        
        breakpt = n - (k%n)
        print(breakpt)
        if(breakpt == n):
            return head
        ptr, prev = head, head
        i = 0
        while i < breakpt:
            i += 1
            prev = ptr
            ptr = ptr.next
        
        tail.next = head
        head = ptr
        prev.next = None
        
        return head